pkg/adapter/kubernetes/blueprintcr/v1/componentDiff.go

```diff
@@ -0,0 +1,91 @@
+package v1
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/cloudogu/cesapp-lib/core"
+
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/adapter/serializer"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain"
+)
+
+// ComponentDiff is the comparison of a Component's desired state vs. its cluster state.
+// It contains the operation that needs to be done to achieve this desired state.
+type ComponentDiff struct {
+	Actual       ComponentDiffState `json:"actual"`
+	Expected     ComponentDiffState `json:"expected"`
+	NeededAction ComponentAction    `json:"neededAction"`
+}
+
+// ComponentDiffState is either the actual or desired state of a component in the cluster.
+type ComponentDiffState struct {
+	Version           string `json:"version,omitempty"`
+	InstallationState string `json:"installationState"`
+}
```

Comment:
Hmm, that is a valid point. The components themselves already come from the OCI registry, so the namespace should be included when downloading. If this is not included here, then that is definitely an error. I shall look into this

- [ ] - done

Reply:
I hope you understood Alex correctly, but we'll ignore namespaces for now.

Reply:
Noooooooo! 
![image](https://github.com/cloudogu/k8s-blueprint-operator/assets/6781173/0c4d081f-9a10-499f-9476-1b0d3b50716e)

Reply:
> I hope you understood Alex correctly, but we'll ignore namespaces for now.

But late @alexander-dammeier wrote
> This is a problem that has been introduced earlier before your PR, but we should really consider adding an extra field for the namespace.

So I included it. What is it now? Are we speaking of the same namespace. For this reason I named the harbor/project name `DistributionNamespace` to have immediately the right mental image of not-talking about k8s namespaces.

Reply:
No sorry, I wrote that, because I didn't know how we planned to handle namespaces. @alexander-dammeier clarified it afterwards.

@alexander-dammeier wrote
> Little background for that:
The plan for now was that we do not handle the cluster namespace for now. Only the default namespace of the component is used. Adding a dedicated field for this in the blueprint was considered a design flaw as the blueprint itself should be independent of kubernetes, so that we don't have the same problems like with packages again.
> 
> the idea:
Components could be a more general concept in blueprints, so that you can use it for any plattform. You just need to add a name, a version and some generic properties (like the k8s-namespace).
> 
> This way you cannot port the blueprint directly to a new plattform as some components will differ but the format could stay the same.

And yes, we're speaking of the same namespace.

Reply:
To conclude: @ppxl and I talked about it and we'll keep the distribution namespace internally but in the blueprint.json it will be part of the component name.